src/server/scripts/Spells/spell_quest.cpp

```diff
@@ -2643,43 +2643,90 @@ class spell_q14100_q14111_make_player_destroy_totems : public SpellScriptLoader
 
 enum Fumping
 {
-    SPELL_SUMMON_SAND_GNOME  = 39240,
-    SPELL_SUMMON_BONE_SLICER = 39241
+    SPELL_SUMMON_SAND_GNOME = 39240,
+    SPELL_SUMMON_SAND_GNOME3 = 39247,
+    SPELL_SUMMON_MATURE_BONE_SIFTER = 39241,
+    SPELL_SUMMON_MATURE_BONE_SIFTER3 = 39245,
+    SPELL_SUMMON_HAISHULUD = 39248
 };
 
 // 39238 - Fumping
-class spell_q10929_fumping : SpellScriptLoader
+class spell_q10929_fumping : public SpellScript
 {
-    public:
-        spell_q10929_fumping() : SpellScriptLoader("spell_q10929_fumping") { }
+    PrepareSpellScript(spell_q10929_fumping);
 
-        class spell_q10929_fumpingAuraScript : public AuraScript
-        {
-            PrepareAuraScript(spell_q10929_fumpingAuraScript);
+    void SetDest(SpellDestination& dest)
+    {
+        Position const offset = { 0.5f, 0.5f, 5.0f, 0.0f };
+        dest.RelocateOffset(offset);
+    }
 
-            bool Validate(SpellInfo const* /*spell*/) override
-            {
-                return ValidateSpellInfo({ SPELL_SUMMON_SAND_GNOME, SPELL_SUMMON_BONE_SLICER });
-            }
+    void Register() override
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_q10929_fumping::SetDest, EFFECT_1, TARGET_DEST_CASTER);
+    }
+};
 
-            void HandleEffectRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
-            {
-                if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
-                    return;
 
-                if (Unit* caster = GetCaster())
-                    caster->CastSpell(caster, urand(SPELL_SUMMON_SAND_GNOME, SPELL_SUMMON_BONE_SLICER), true);
-            }
+class spell_q10929_fumpingAuraScript : public AuraScript
+{
+    PrepareAuraScript(spell_q10929_fumpingAuraScript);
 
-        void Register() override
-        {
-            OnEffectRemove += AuraEffectRemoveFn(spell_q10929_fumpingAuraScript::HandleEffectRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
-        }
-    };
+    bool Validate(SpellInfo const* /*spell*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SUMMON_SAND_GNOME,SPELL_SUMMON_SAND_GNOME3, SPELL_SUMMON_MATURE_BONE_SIFTER });//changed 24042021
```

Comment:
we usually don't include comments about when something was changed, there is `git blame` for that

Reply:
```suggestion
        return ValidateSpellInfo({ SPELL_SUMMON_SAND_GNOME,SPELL_SUMMON_SAND_GNOME3, SPELL_SUMMON_MATURE_BONE_SIFTER });
```